docs/source/download.rst

```diff
@@ -98,7 +98,7 @@ packages can be installed into an existing Firedrake installation using
 System requirements
 -------------------
 
-Firedrake requires Python 3.6 or later. The installation script is
+Firedrake requires Python 3.6.x to 3.8.x. The installation script is
```

Comment:
Add a note about lack of binary wheel for Pyhton 3.9 packages?

Reply:
Changed in rebased commit 9381afe